hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -521,21 +568,52 @@ protected List<RegionPlan> balanceTable(TableName tableName,
     // Perform a stochastic walk to see if we can get a good fit.
     long step;
 
+    boolean improvedConditionals = false;
+    Map<Class<? extends CandidateGenerator>, Long> generatorToStepCount = new HashMap<>();
+    Map<Class<? extends CandidateGenerator>, Long> generatorToApprovedActionCount = new HashMap<>();
```

Comment:
These aren't really necessary, but make debugging easier because you can understand which candidate generators lead to the given balancer plan. If we had this logging, I think we would've realized the flaws in getRandomGenerator earlier (it being prone to just picking the last generator in the list)